lib/src/wip/autocomplete/popup.dart

```diff
@@ -50,13 +53,13 @@ class _PopupState extends State<Popup> {
 
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: EdgeInsets.only(
-        left: min(widget.column, widget.editingWindowSize.width - width),
-        top: widget.row,
-      ),
-      child: ConstrainedBox(
+    return Positioned(
+      left: min(widget.column, widget.editingWindowSize.width - width),
```

Comment:
The problem here is that this blurs the responsibility of positioning. The positioning should be one in one of these ways:

1. The outer widget determines the exact position and only passes the offset, and not alignment or window size.
2. The outer widget passes the caret position and the window size and lets this widget decide on the rest.

The 2nd one follows the SRP closer because the field has less reasons to change.

Reply:
Also offsets should normally not be passed as separate coördinates because it means twice the job to handle. I suggest you make a single property for `x` and `y`. It can be `offset` if you let the outer widget position the popup or `preferredOffset` if you delegate this to the popup. Flutter has the `Offset` class for this.